datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -388,28 +391,82 @@ struct RowGroupPruningStatistics<'a> {
     parquet_schema: &'a Schema,
 }
 
+// TODO: consolidate code with arrow-rs
+// Convert the bytes array to i128.
+// The endian of the input bytes array must be big-endian.
+// Copy from the arrow-rs
+fn from_bytes_to_i128(b: &[u8]) -> i128 {
```

Comment:
Would it be possible to simply call `Decimal128::new()` here? https://docs.rs/arrow/19.0.0/arrow/util/decimal/struct.Decimal128.html#method.new

Reply:
`Decimal128::new()` can't be apply to this, because the length of [u8] must be 16 and with default little-endian.
The array of [u8] decoded from parquet may be not align with 16 bytes, and the endian of it is big-endian.
About the definition of parquet decimal data, you can read the [format spec](https://github.com/apache/parquet-format/blob/54e53e5d7794d383529dd30746378f19a12afd58/src/main/thrift/parquet.thrift#L66)